src/func.cc

```diff
@@ -656,6 +657,59 @@ void ShellFuncNoRerun(const std::vector<Value*>& args,
   ShellStatusVar::SetValue(returnCode);
 }
 
+void VarVisibilityFunc(const std::vector<Value*>& args,
+                       Evaluator* ev,
+                       std::string* s) {
+  std::string arg = args[0]->Eval(ev);
+  std::vector<std::string> prefixes;
+
+  std::stringstream ss(args[1]->Eval(ev));
+  std::string prefix;
+  while (ss >> prefix) {
+    if (HasPrefix(prefix, "/")) {
+      ERROR_LOC(ev->loc(), "Visibility prefix should not start with /");
+    }
+    if (HasPrefix(prefix, "../")) {
+      ERROR_LOC(ev->loc(), "Visibility prefix should not start with ../");
+    }
+
+    std::string normalizedPrefix = prefix;
+    NormalizePath(&normalizedPrefix);
+    if (prefix != normalizedPrefix) {
+      ERROR_LOC(ev->loc(),
+                "Visibility prefix %s is not normalized. Normalized prefix: %s",
+                prefix.c_str(), normalizedPrefix.c_str());
+    }
+
+    // one visibility prefix cannot be the prefix of another visibility prefix
+    for (std::vector<std::string>::iterator it = prefixes.begin();
+         it != prefixes.end(); ++it) {
+      if (HasPrefix(*it, prefix)) {
+        ERROR_LOC(ev->loc(),
+                  "Visibility prefix %s is the prefix of another visibility "
+                  "prefix %s",
+                  prefix.c_str(), it->c_str());
+      } else if (HasPrefix(prefix, *it)) {
```

Comment:
These also need to use path-like prefixes instead of string-like prefixes. Consider adding a test where you have visibility to both "ven" and "vendor".

Since you'll now have 3 places where you use path-link prefixes, consider making a HasPathPrefix helper function.